frog/imports/ui/StudentView.jsx

```diff
@@ -108,6 +109,62 @@ const StudentView = ( { user, sessions, products } ) => {
   </div>
 )}
 
+var TimerExample = React.createClass({
+
+    getInitialState: function(){
+
+        // This is called before our render function. The object that is 
+        // returned is assigned to this.state, so we can use it later.
+
+        return { elapsed: 0 };
+    },
+
+    componentDidMount: function(){
+
+        // componentDidMount is called by react when the component 
+        // has been rendered on the page. We can set the interval here:
+
+        this.timer = setInterval(this.tick, 50);
+    },
+
+    componentWillUnmount: function(){
+
+        // This method is called immediately before the component is removed
+        // from the page and destroyed. We can clear the interval here:
+
+        clearInterval(this.timer);
+    },
+
+    tick: function(){
+
+        // This function is called every 50 ms. It updates the 
+        // elapsed counter. Calling setState causes the component to be re-rendered
+
+        this.setState({elapsed: new Date() - this.props.start});
+    },
```

Comment:
Although given that we might have several components that need to show clock information, it would be interesting to write a higher-level function which takes a React component and makes it regularly update. Not a priority, but I might play with this :)